src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/VersionConverter.cs

```diff
@@ -37,38 +34,51 @@ private static Version ReadCore(ref Utf8JsonReader reader)
             Debug.Assert(reader.TokenType is JsonTokenType.PropertyName or JsonTokenType.String);
 
 #if NET
-            if (!JsonHelpers.IsInRangeInclusive(reader.ValueLength, MinimumVersionLength, MaximumEscapedVersionLength))
+            int bufferLength = reader.ValueLength;
+            char[]? rentedBuffer = null;
+            try
             {
-                ThrowHelper.ThrowFormatException(DataType.Version);
+                Span<char> charBuffer = bufferLength <= JsonConstants.StackallocCharThreshold
+                    ? stackalloc char[JsonConstants.StackallocCharThreshold]
+                    : (rentedBuffer = ArrayPool<char>.Shared.Rent(bufferLength));
+
+                int bytesWritten = reader.CopyString(charBuffer);
+                ReadOnlySpan<char> source = charBuffer.Slice(0, bytesWritten);
+
+                if (source.IsEmpty || char.IsWhiteSpace(source[0]) || char.IsWhiteSpace(source[^1]))
+                {
+                    // Since leading and trailing whitespaces are forbidden throughout System.Text.Json converters
+                    // we need to make sure that our input doesn't have them,
+                    // and if it has - we need to throw, to match behaviour of other converters
+                    // since Version.TryParse allows them and silently parses input to Version
+                    ThrowHelper.ThrowFormatException(DataType.Version);
+                }
+
+                bool success = Version.TryParse(source, out Version? result);
```

Comment:
.NET 10 has an overload of Version.TryParse that operates on a `ReadOnlySpan<byte>` of UTF8 bytes. When on .NET 10+ and the Utf8JsonReader contains the whole value buffered in a span, could this parse directly from that, rather than needing to get a temporary buffer, transcode into it, and then parse that?